### PR TITLE
add trigger of deployment off of instantiate endpoint

### PR DESCRIPTION
**Pull request by gabemontero**
@jcantrill PTAL

### CODE REVIEW

**Comment by gabemontero** [comment]
[test]

File: src/main/java/com/openshift/internal/restclient/model/deploy/DeploymentRequest.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+import java.util.Map;
+
+import org.jboss.dmr.ModelNode;
+
+import com.openshift.internal.restclient.model.KubernetesResource;
+import com.openshift.restclient.IClient;
+import com.openshift.restclient.model.deploy.IDeploymentRequest;
+
+public class DeploymentRequest extends KubernetesResource implements IDeploymentRequest {
```
Comment (jcantrill) [inline src/main/java/com/openshift/internal/restclient/model/deploy/DeploymentRequest.java]: We didnt do this with BuildRequest but probably should have.  I tried to separate by:  'KubernetesResource' is for top level objects that actually have an endpoint vs parameter type objects which should use ModelNodeAdapter.  Please try to refactor.  If it becomes too large of an issue, not that important.

File: src/main/java/com/openshift/internal/restclient/model/deploy/DeploymentRequest.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    
+    private static final String LATEST = "latest";
+    private static final String FORCE = "force";
+    // the DeploymentRequest has the name at the top level vs. the k8s metadata struct
+    private static final String NAME = "name";
```
Comment (jcantrill) [inline src/main/java/com/openshift/internal/restclient/model/deploy/DeploymentRequest.java]: comes in as part of ResourcePropertyKeys

File: src/main/java/com/openshift/internal/restclient/model/deploy/DeploymentRequest.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+        super(node, client, overrideProperties);
+    }
+
+    @Override
+    public void setLatest(boolean latest) {
+        set(LATEST, latest);
+    }
+
+    @Override
+    public boolean getLatest() {
```
Comment (jcantrill) [inline src/main/java/com/openshift/internal/restclient/model/deploy/DeploymentRequest.java]: convention generally is 'isLatest()'

File: src/main/java/com/openshift/internal/restclient/model/deploy/DeploymentRequest.java [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+        return get(LATEST).asBoolean();
+    }
+
+    @Override
+    public void setForce(boolean force) {
+        set(FORCE, force);
+    }
+
+    @Override
+    public boolean getForce() {
```
Comment (jcantrill) [inline src/main/java/com/openshift/internal/restclient/model/deploy/DeploymentRequest.java]: isForce()

File: src/main/java/com/openshift/internal/restclient/model/deploy/DeploymentRequest.java (hunk 3 above)
Comment (gabemontero) [inline src/main/java/com/openshift/internal/restclient/model/deploy/DeploymentRequest.java]: doh - yep, forgot

File: src/main/java/com/openshift/internal/restclient/model/deploy/DeploymentRequest.java (hunk 4 above)
Comment (gabemontero) [inline src/main/java/com/openshift/internal/restclient/model/deploy/DeploymentRequest.java]: ditto

File: src/main/java/com/openshift/internal/restclient/model/deploy/DeploymentRequest.java (hunk 1 above)
Comment (gabemontero) [inline src/main/java/com/openshift/internal/restclient/model/deploy/DeploymentRequest.java]: I have this refactor working.  Will push soon.

**Comment by gabemontero** [comment]
updates based on comments pushed

[test]

**Comment by jcantrill** [comment]
closing in favor of #230

**Pull request closed without merging**